apps/els_dap/src/els_dap_general_provider.erl

```diff
@@ -425,39 +447,73 @@ handle_info( {int_cb, ThreadPid}
             , frames => stack_frames(ThreadPid, ProjectNode)
             },
   {Module, Line} = break_module_line(ThreadPid, ProjectNode),
-
-  %% handle breakpoints
-  Mode1 =
-    case els_dap_breakpoints:type(Breakpoints, Module, Line) of
-      regular ->
-        els_dap_server:send_event( <<"stopped">>
-                                 , #{ <<"reason">> => <<"breakpoint">>
-                                    , <<"threadId">> => ThreadId
-                                    }),
-        stepping;
-      {log, Expression} ->
-        Return = safe_eval(ProjectNode, ThreadPid, Expression, no_update),
-        LogMessage = unicode:characters_to_binary(
-          io_lib:format("~s:~b - ~w~n",
-                        [source(Module, ProjectNode), Line, Return])
-        ),
-        els_dap_server:send_event( <<"output">>
-                                 , #{ <<"output">> => LogMessage }),
-        case Mode0 of
-          running ->
-            els_dap_rpc:continue(ProjectNode, ThreadPid);
+  Breakpt = els_dap_breakpoints:type(Breakpoints, Module, Line),
+  %% evaluate condition if exists, otherwise treat as 'true'
+  Condition = case Breakpt of
```

Comment:
Can we break the implementation of the handle_info clause into smaller functions? It's 100 loc for one clause currently. Each of the blocks could be a sub-function

Reply:
Thanks for the feedback! Yes, that makes sense, will do.